ios/Classes/HyperTrackPlugin.m

```diff
@@ -120,78 +128,54 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
         result(nil);
         return;
     }
-
     
-    result(FlutterMethodNotImplemented);
-}
-
-- (void)onSdkError:(NSNotification*)notification {
-    NSLog(@"%s notification %@", __PRETTY_FUNCTION__, notification);
-    NSError* error = [notification hyperTrackTrackingError];
-    if (error == nil) return;
-
-    NSLog(@"Got error %@", error);
-    _eventSink([self convertErrorToMessage:error]);
+    if ([@"isTracking" isEqualToString:call.method]) {
+        result([NSNumber numberWithBool:[self.hyperTrack isTracking]]);
+        return;
+    }
     
+    if([@"getLatestLocation" isEqualToString:call.method]) {
+        result([self.hyperTrack location]);
+        return;
+    }
+    
+    if([@"setAvailability" isEqualToString:call.method]) {
+        NSNumber *available = call.arguments;
+        [self.hyperTrack setAvailability:(available.boolValue ? HTAvailabilityAvailable : HTAvailabilityUnavailable)];
+        result(nil);
+        return;
+    }
+    
+    if([@"getAvailability" isEqualToString:call.method]) {
+        if([self.hyperTrack availability] == HTAvailabilityAvailable) {
+            result(@"available");
+        } else {
+            result(@"unavailable");
```

Comment:
why do we pass a string here? and in the availability stream handler? IMO strings are leaky and prone to typos. it's a "isAvailable => true/false" value. let's pass a boolean.

Reply:
IMO it's better to have string there because Availability at thhe SDK API level is not a boolean, but enum, so it's harder to mess things up this way (you can always know the datatype from the value)